quadriga/src/main/java/edu/asu/diging/quadriga/aspect/AuthAspect.java

```diff
@@ -0,0 +1,54 @@
+package edu.asu.diging.quadriga.aspect;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+
+import edu.asu.diging.quadriga.config.web.TokenInfo;
+
+@Aspect
+@Configuration
+public class AuthAspect {
+
+    @Pointcut("execution(@edu.asu.diging.quadriga.aspect.annotation.InjectToken * *(..))")
+    public void annotatedMethod() {}
+
+    @Pointcut("execution(* *(.., edu.asu.diging.quadriga.config.web.TokenInfo, ..))")
+    public void tokenParamMethod() {}
+
+    @Around("annotatedMethod() && tokenParamMethod()")
+    public Object authAdvice(ProceedingJoinPoint joinPoint) throws Throwable {
+        Object[] args = joinPoint.getArgs();
+        int authIndex = getIndexOf(args, Authentication.class);
+        int tokenIndex = getIndexOf(args, TokenInfo.class);
+
+        if (tokenIndex >= 0) {
+            args[tokenIndex] = null;
+            if (authIndex >= 0) {
```

Comment:
actually, I think in this case I would use the `SecurityContextHolder` to get the current authentication. Then you don't have to search through the arguments and it's one thing less the developer has to remember.